storage/s3/src/test/java/io/aiven/kafka/tieredstorage/storage/s3/S3ClientBuilderTest.java

```diff
@@ -45,6 +47,9 @@ void minimalConfig() {
             .allSatisfy(metricPublisher -> assertThat(metricPublisher).isInstanceOf(MetricCollector.class));
         assertThat(clientConfiguration.overrideConfiguration().apiCallTimeout()).isEmpty();
         assertThat(clientConfiguration.overrideConfiguration().apiCallAttemptTimeout()).isEmpty();
+        assertThat(clientConfiguration.overrideConfiguration().toBuilder().executionInterceptors())
+            .extracting(ExecutionInterceptor::getClass)
+            .doesNotHaveAnyElementsOfTypes(LegacyMd5ExecutionInterceptor.class);
```

Comment:
How this works if by default the config is `false`?

Reply:
What do you mean? The assertion checks that by default it is disabled.